spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/TestCompressionSettings.java

```diff
@@ -108,14 +108,14 @@ public static Object[][] parameters() {
         SparkCatalogConfig.SPARK.implementation(),
         SparkCatalogConfig.SPARK.properties(),
         PARQUET,
-        ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_LEVEL, "1")
```

Comment:
nit: We are only partially alphabetizing here

This is the kind of thing I do love but it should be

gzip
snappy
zstd

I would probably just skip this change for now and do it in another PR

Reply:
The reason I wanted to `ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_LEVEL, "1")` after gzip is that the new Hadoop version uses `CompressionLevel` to initialize a GzipCompressor, and this COMPRESSION_LEVEL, "1", is carried over to gzip. However, "1" is not a valid compression level for gzip, so it throws an exception.
```
    org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 6.0 failed 1 times, most recent failure: Lost task 1.0 in stage 6.0 (TID 7) (192.168.50.141 executor driver): java.lang.IllegalArgumentException: No enum constant org.apache.hadoop.io.compress.zlib.ZlibCompressor.CompressionLevel.1
    	at java.base/java.lang.Enum.valueOf(Enum.java:273)
    	at org.apache.hadoop.conf.Configuration.getEnum(Configuration.java:1786)
    	at org.apache.hadoop.io.compress.zlib.ZlibFactory.getCompressionLevel(ZlibFactory.java:165)
    	at org.apache.hadoop.io.compress.zlib.BuiltInGzipCompressor.init(BuiltInGzipCompressor.java:157)
    	at org.apache.hadoop.io.compress.zlib.BuiltInGzipCompressor.<init>(BuiltInGzipCompressor.java:67)
    	at org.apache.hadoop.io.compress.GzipCodec.createCompressor(GzipCodec.java:64)
    	at org.apache.hadoop.io.compress.CodecPool.getCompressor(CodecPool.java:152)
    	at org.apache.hadoop.io.compress.CodecPool.getCompressor(CodecPool.java:168)
    	at org.apache.parquet.hadoop.CodecFactory$HeapBytesCompressor.<init>(CodecFactory.java:157)
    	at org.apache.parquet.hadoop.CodecFactory.createCompressor(CodecFactory.java:219)
    	at org.apache.parquet.hadoop.CodecFactory.getCompressor(CodecFactory.java:202)
    	at org.apache.iceberg.parquet.ParquetWriter.<init>(ParquetWriter.java:90)
    	at org.apache.iceberg.parquet.Parquet$WriteBuilder.build(Parquet.java:360)
```
I think this over; rather than switching the order, it's better to unset the COMPRESSION_CODEC and COMPRESSION_LEVEL for each test.

Reply:
I probably should have a separate PR and fix this in Spark3.5 too. WDYT?